.github/workflows/pytest.yaml

```diff
@@ -43,7 +43,37 @@ jobs:
             package_list.txt
             pip_list.txt
 
+      - name: Clone caret_analyze_cpp_impl
+        run: |
+          git clone https://github.com/tier4/caret_analyze_cpp_impl.git
+          cd caret_analyze_cpp_impl/
+          mv CARET_analyze_cpp_impl src
+
+      - name: Restore Cache
+        id: cpp_impl-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: caret_analyze_cpp_impl/install
+          key: ${{ runner.os }}-node-cpp_impl-${{ hashFiles('caret_analyze_cpp_impl/src/**') }}
+
+      - name: Build caret_analyze_cpp_impl
+        if: steps.cpp_impl-cache.outputs.cache-hit != 'true'
+        run: |
+          source /opt/ros/humble/setup.bash
+          cd caret_analyze_cpp_impl/
+          colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
+        shell: bash
+
+      - name: Cache caret_analyze_cpp_impl
+        if: steps.cpp_impl-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: caret_analyze_cpp_impl/install
+          key: ${{ runner.os }}-node-cpp_impl-${{ hashFiles('caret_analyze_cpp_impl/src/**') }}
+
       - name: Run pytest
         run: |
+          mv caret_analyze_cpp_impl/install /__w
```

Comment:
The reason for moving the directory is to shorten the PYTHONPATH. If PYTHONPATH is longer than 99 characters, it violates the code conventions flake8 and pytest will not pass.

Reply:
If you are using colcon build, install/setup.bash should work fine.
Or you can simply break the line.

Reply:
Change the method of environment configuration.

Reply:
I have confirmed it in [b07cfe0](https://github.com/tier4/caret_analyze/pull/405/commits/b07cfe049efc3c3dd4a05fe8569453c8cc3fd611).